apiml-security-common/src/main/java/org/zowe/apiml/security/common/audit/RauditxService.java

```diff
@@ -0,0 +1,368 @@
+ * On the initialization of this bean the privileges are checked and could write a warning message in the console log.
+ *
+ * In case this feature is not available (not enough credentials, the service runs off z/OS) it will not throw any
+ * exception. The code should not check the possibility to issue SMF record neither.
+ *
+ * Example:
+ *
+ * <pre>
+ * &#64;Service
+ * &#64;RequiredArgsConstructor
+ * class AuditedClass {
+ *
+ *  private final RauditxService rauditxService;
+ *
+ *  void doSomething() {
+ *      ...
+ *      rauditxService.builder().
+ *          .userId("userId")
+ *          .messageSegment("An attempt to generate PAT")
+ *          .alwaysLogSuccesses()
+ *          .alwaysLogFailures()
+ *          .issue();
+ *      ...
+ *  }
+ *
+ * }
+ * </pre>
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class RauditxService {
+
+    // documented types at https://www.ibm.com/docs/en/zos/2.2.0?topic=records-smf-record-type-83-subtype-2
+    private static final int RELOCATED_RECORD_TYPE_BIND_USER = 103;
+
+    @Value("${rauditx.fmid:AZWE001}")
+    private String fmid;
+
+    @Value("${rauditx.component:ZOWE}")
+    private String component;
+
+    // Description of subtypes at https://www.ibm.com/docs/en/zos/2.5.0?topic=records-record-type-83-security-events
+    @Value("${rauditx.subtype:2}")
+    private int subtype;
+
+    // Events and qualifiers documentation at https://www.ibm.com/docs/en/zos/2.5.0?topic=descriptions-event-codes-event-code-qualifiers
+    @Value("${raudit.event:2}")
+    private int event;
+
+    @Value("${rauditx.qualifier.success:0}")
+    private int qualifierSuccess;
+
+    @Value("${rauditx.qualifier.failed:1}")
+    private int qualifierFailed;
+
+    String getCurrentUser() {
+        try {
+            Class<?> zutilClass = Class.forName("com.ibm.jzos.ZUtil");
+            MethodHandle getCurrentUser = MethodHandles.publicLookup().findStatic(zutilClass, "getCurrentUser", MethodType.methodType(String.class));
+            return (String) getCurrentUser.invoke();
+        } catch (Throwable t) {
+            log.debug("Cannot obtain current userId", t);
+            return null;
+        }
+    }
+
+    void logNoPrivileges(String userId) {
+        if (StringUtils.isBlank(userId)) {
+            log.debug("Cannot issue any Rauditx record off z/OS.");
+        } else {
+            log.warn("The calling userid ({}) must have READ authority to the IRR.RAUDITX profile in the FACILITY class to issue a Rauditx record.", userId);
+        }
+    }
+
+    SafResourceAccessVerifying getNativeSafResourceAccessVerifying() {
+        try {
+            return new SafResourceAccessSaf();
+        } catch (Exception e) {
+            log.debug("Cannot create instance of SafResourceAccessSaf");
+            return null;
+        }
+    }
+
+    @PostConstruct
+    public void verifyPrivileges() {
+        String userId = getCurrentUser();
+        boolean hasAccess = false;
```

Comment:
Can we use the user from configuration for this?

Reply:
Maybe, which do you think? ZWE_zowe_setup_security_users_zowe? Anyway, this uses the right one and it is independent of configuration.